src/System.Management.Automation/namespaces/FileSystemProvider.cs

```diff
@@ -123,28 +123,29 @@ private static string GetCorrectCasedPath(string path)
             {
                 string exactPath = string.Empty;
                 int itemsToSkip = 0;
-                if (Utils.PathIsUnc(path))
+                if (Utils.PathIsUnc(path) || path.StartsWith("\\\\?\\"))
```

Comment:
Oh, I am not like the change because it is seem _partial_ - general PowerShell design is to bypass paths started with `\\?\`. So I'd expect the check is done in PathIsUnc() but the last method is used 17 times and there are contradictions. For example, there is IsValidPath() method and PathIsNetworkPath(). 
Also I see we never test and review all scenarios with `\\?\`.

Notice, cmd.exe is not support the scenario too
```
C:\>dir \\?\Volume{0qq60da1-0000-0000-0000-100000000000}\
The parameter is incorrect.
```

Reply:
I originally thought about putting it in `PathIsUnc()`, but it's technically not UNC so didn't.  However, I don't think there's anything functionally different from a UNC path that just happens to point to a local location, so it probably makes sense to encapsulate it there.

Reply:
Looks like we already handle `wsl$` path in `PathIsUnc()` so makes even more sense to have it there